docs/root/configuration/http/http_filters/oauth2_filter.rst

```diff
@@ -0,0 +1,71 @@
+
+.. _config_http_filters_oauth:
+
+OAuth2
+======
+
+* :ref:`v3 API reference <envoy_v3_api_msg_extensions.filters.http.oauth2.v3.OAuth2>`
+* This filter should be configured with the name *envoy.filters.http.oauth2*.
+
+.. attention::
+
+  The OAuth2 filter is currently under active development.
+
+Example configuration
+---------------------
+
+.. code-block::
+
+   http_filters:
+   - name: oauth2
+     typed_config:
+       "@type": type.googleapis.com/envoy.extensions.filters.http.oauth2.v3.OAuth2
+       cluster: oauth
+      hostname: oauth.com
+      callback_path: /callback
+      signout_path: /signout
+      token_path: /token
+      credentials:
+        client_id: foo
+        token_secret:
+          name: token
```

Comment:
Presumably these SDS secret configs also reference `auth` cluster?

Reply:
I specified them as static SDS secrets, but if you think it would be useful I can make them dynamic and reference an SDS cluster?